react-hot-button-component/src/index.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { clicks: 0 };
+    this.handleClick = this.handleClick.bind(this);
+    this.counter = 0;
+
+  }
+
+  handleClick() {
+    this.setState({ clicks: this.counter++ });
+    this.counter++;
+  }
+
+  render() {
+
+    let classes;
+    // eslint-disable-next-line react/prop-types
+
+    if (this.counter >= 3 && this.counter < 6) {
+      classes = 'lowest';
+    } else if (this.counter >= 6 && this.counter < 9) {
+      classes = 'low';
+    } else if (this.counter >= 9 && this.counter < 12) {
+      classes = 'forgotten';
+    } else if (this.counter >= 12 && this.counter < 15) {
+      classes = 'medium';
+    } else if (this.counter >= 15 && this.counter < 18) {
+      classes = 'hot';
+    } else if (this.counter >= 18) {
+      classes = 'hottest';
+    }
```

Comment:
Here you should be using the value from state, state is the single source of truth, it should be the only value that is relied on when determining what should be rendered.